micronota/parsers/cmscan.py

```diff
@@ -0,0 +1,413 @@
+17. inc
+   Indicates whether or not this hit achieves the inclusion threshold: ’!’ if
+   it does, ’?’ if it does not (and rather only achieves the reporting
+   threshold). By default, the inclusion threshold is an E-value of 0.01 and
+   the reporting threshold is an E-value of 10.0, but these can be changed with
+   command line options as described in the manual pages.
+18. description of target
+   The remainder of the line is the target’s description line, as free text.
+
+
+Format Support
+--------------
+**Has Sniffer: No**
+
+
+Reference
+---------
+.. [1] Eric P. Nawrocki and Sean R. Eddy, "Infernal 1.1: 100-fold faster RNA
+       homology searches",  Bioinformatics 2013,
+       doi: 10.1093/bioinformatics/btt509
+
+'''
+
+from skbio.io import create_format, FileFormatError
+from skbio.metadata import IntervalMetadata, Feature
+from skbio.io.format._base import (_line_generator, _too_many_blanks)
+from skbio.io.format._base import _get_nth_sequence as _get_nth_record
+
+cmscan = create_format('cmscan')
+
+# column headers
+_COLUMNS = {
+   'MODEL_NAME':              {'position':  0, 'width': 22, 'textalignment': 'l', 'caption': 'target name'},
+   'MODEL_ACCESSION':         {'position':  1, 'width':  9, 'textalignment': 'l', 'caption': 'accession'},
+   'SEQUENCE_NAME':           {'position':  2, 'width': 28, 'textalignment': 'l', 'caption': 'query name'},
+   'SEQUENCE_ACCESSION':      {'position':  3, 'width':  9, 'textalignment': 'l', 'caption': 'accession'},
+   'TYPE_OF_MODEL':           {'position':  4, 'width':  3, 'textalignment': 'r', 'caption': 'mdl'},
+   'MODEL_START_POSITION':    {'position':  5, 'width':  8, 'textalignment': 'r', 'caption': 'mdl from'},
+   'MODEL_END_POSITION':      {'position':  6, 'width':  8, 'textalignment': 'r', 'caption': 'mdl to'},
+   'SEQUENCE_START_POSITION': {'position':  7, 'width':  8, 'textalignment': 'r', 'caption': 'seq from'},
+   'SEQUENCE_END_POSITION':   {'position':  8, 'width':  8, 'textalignment': 'r', 'caption': 'seq to'},
+   'STRAND':                  {'position':  9, 'width':  6, 'textalignment': 'r', 'caption': 'strand'},
+   'TRUNCATED':               {'position': 10, 'width':  5, 'textalignment': 'r', 'caption': 'trunc'},
+   'PASS':                    {'position': 11, 'width':  4, 'textalignment': 'r', 'caption': 'pass'},
+   'GC_CONTENT':              {'position': 12, 'width':  4, 'textalignment': 'r', 'caption': 'gc'},
+   'BIAS':                    {'position': 13, 'width':  5, 'textalignment': 'r', 'caption': 'bias'},
+   'BITSCORE':                {'position': 14, 'width':  6, 'textalignment': 'r', 'caption': 'score'},
+   'EVALUE':                  {'position': 15, 'width':  9, 'textalignment': 'r', 'caption': 'E-value'},
+   'INC':                     {'position': 16, 'width':  3, 'textalignment': 'l', 'caption': 'inc'},
+   'DESCRIPTION':             {'position': 17, 'width': 21, 'textalignment': 'l', 'caption': 'description of target'}
+}
```

Comment:
I removed all functions and attributes related to writing capabilities.
Since writers are gone, we don't care about positions (btw. they are fixed). Thus, _COLUMN is now much slimmer.